dlt/extract/extract.py

```diff
@@ -80,8 +85,7 @@ def _write_item(table_name: str, item: TDataItems) -> None:
             collector.update(table_name)
```

Comment:
1. you should track the resource names that generated data, not table names 
2. this is the only place that you should update the list. it is called in every case
3. pass both table_name and resource_name here, use table name for collector, resource name to track your set of resources

Reply:
see below why we track resources, not tables